src/solenoid.cpp

```diff
@@ -0,0 +1,24 @@
+#include "solenoid.h"
+
+Solenoid::Solenoid(int pin) : pin_(pin) { plunger_state_ = kRetracted; }
+
+Solenoid::~Solenoid() {
+  // Retract solenoid upon destruction
+  retract();
+}
+
+void Solenoid::extend() {
+  if (plunger_state_ != kExtended) {
+    // TODOLater - implement actuating solenoid
```

Comment:
The datasheet for this hardware is very sparse, so there's no real way of telling how to actuate the valve with the Arduino without having it in our hands